### PR TITLE
fixie auth pull in a template file if entry_point doesn't exist

### PR DESCRIPTION
**Pull request by hessamb**

### CODE REVIEW

File: fixieai/cli/agent/commands.py [hunk 1]
```diff
@@ -14,6 +16,14 @@
 from fixieai.cli.agent import agent_config
 from fixieai.cli.agent import loader
 
+# Regex pattern to match valid entry points: "module:object"
+VAR_NAME_RE = r"(?![0-9])\w+"
+ENTRY_POINT_PATTERN = re.compile(rf"^{VAR_NAME_RE}(\.{VAR_NAME_RE})*:{VAR_NAME_RE}$")
+# The agent template main.py file
+AGENT_TEMPLATE_URL = (
+    "https://raw.githubusercontent.com/fixie-ai/fixie-examples/main/agents/template.py"
```
Comment (juberti) [inline fixieai/cli/agent/commands.py:24]: as discussed let's remove everything but the basics from the template

File: fixieai/cli/agent/commands.py [hunk 2]
```diff
@@ -81,6 +101,17 @@ def init_agent(handle, description, entry_point, more_info_url, public):
     current_config.public = public
     agent_config.save_config(current_config)
 
+    entry_module, _ = entry_point.split(":")
+    expected_main_path = entry_module.replace(".", "/") + ".py"
+    if not os.path.exists(expected_main_path):
+        urllib.request.urlretrieve(AGENT_TEMPLATE_URL, expected_main_path)
+        click.secho(
+            f"Initialized agent.yaml and made a template agent file at {expected_main_path}",
```
Comment (juberti) [inline fixieai/cli/agent/commands.py:109]: maybe "main.py" instead of "agent" to make clear it's python?

**Review by juberti — APPROVED** [review]
LG with nit

File: fixieai/cli/agent/commands.py (hunk 2 above)
Comment (hessamb) [reply fixieai/cli/agent/commands.py:109]: The default file is `main.py` (unless the user changes the entry point in the questionnaire). The `agent.yaml` holds the config

File: fixieai/cli/agent/commands.py (hunk 1 above)
Comment (hessamb) [reply fixieai/cli/agent/commands.py:24]: Will do in the fixie example repo

**Pull request merged**